src/actions/marketing-actions.js

```diff
@@ -211,9 +211,19 @@ export const saveMarketingSetting =
   async (dispatch, getState) => {
     if (entity.type === MARKETING_SETTING_TYPE_FILE && !file)
       return Promise.resolve();
-    // TODO: review to save hex color settings with empty values
-    if (entity.type === MARKETING_SETTING_TYPE_HEX_COLOR && !entity.value)
+    if (
```

Comment:
could we refactor this code to something like

// Helper function to check if the entity is a hex color setting
if (isHexColorSetting(entity)) {
  // If entity is new and has no value , dont save it
  if (!entity.id && !entity.value) return Promise.resolve();

  // if entity exists and has no value, delete it
  if (entity.id && !entity.value) return dispatch(deleteSetting(entity.id));
}